odh-notebook-controller/odh-notebook-controller/manager/manager.yaml

```diff
@@ -22,10 +22,11 @@ spec:
       containers:
         - name: manager
           image: quay.io/opendatahub/odh-notebook-controller:latest
-          imagePullPolicy: Always
+          imagePullPolicy: IfNotPresent
           command:
             - /manager
-          args: ["--oauth-proxy-image", "registry.redhat.io/openshift4/ose-oauth-proxy:v4.10"]
+          # latest v4.10 manifest link digest for architecture AMD64. Used instead of tag format to be compatible with imagePullPolicy: IfNotPresent in ose-oauth sidecars
+          args: ["--oauth-proxy-image", "registry.redhat.io/openshift4/ose-oauth-proxy@sha256:ab112105ac37352a2a4916a39d6736f5db6ab4c29bad4467de8d613e80e9bb33"]
```

Comment:
Are you sure we need to use manifest list digest and not image digest itself?
The image digest seems to be different from the catalog. 
Observed it also on https://github.com/opendatahub-io/kubeflow/pull/113

Reply:
Yes, manifest list digest is the digest to pull. It contains infos in layers, location repo and so on. You can pull an Image by tag e.g. v4.10 and you get back what is referred to as manifest list digest. If you docker inspect an Image pulled by tag, you see sha256 Image id section and manifest link section.

See also https://www.opensourcerers.org/2020/11/16/container-images-multi-architecture-manifests-ids-digests-whats-behind/

Reply:
Another link and explanation saying the same

https://learn.microsoft.com/en-us/azure/container-registry/container-registry-concepts

To address a registry artifact for push and pull operations with Docker or other client tools, combine the fully qualified registry name, repository name (including namespace path if applicable), and an artifact tag or manifest digest. 

Manifest digest

Manifests are identified by a unique SHA-256 hash, or manifest digest. Each image or artifact--whether tagged or not--is identified by its digest. The digest value is unique even if the artifact's layer data is identical to that of another artifact. This mechanism is what allows you to repeatedly push identically tagged images to a registry. For example, you can repeatedly push myimage:latest to your registry without error because each image is identified by its unique digest.

Reply:
Maybe e.g. @Jooho can also give input.

I went to 
https://catalog.redhat.com/software/containers/openshift4/ose-oauth-proxy/5cdb2133bed8bd5717d5ae64?architecture=amd64&tag=v4.10.0-202306170106.p0.g799d414.assembly.stream&push_date=1688610772000&container-tabs=gti sesrched for v4.10 tag and the under tag Get this image used the Manifest List Digest as of July 6.

Reply:
Great, and thank you for providing the references! I am ok with the manifests digest; I just wanted to confirm which of the two options is more appropriate.

Reply:
While we should be updating the `ose-oauth-proxy` image at a regular cadence,  we should not be hardcoding the image digest into the container command argument of the `Deployment` object.  This should be replaced with an environment variable that can be configured at when the kustomize manifest is parsed.  This would also allow us to replace with as a CSV `relatedImage` or modified by the new operator.

Reply:
I kept this off the table in this PR as this is more a a conceptual change that the component makers should make. The reason I added digest is to be IfNotPresent compatible for a regularly changing image behind tag v4.10.

Reply:
Let me remove that particular change (digest value ose-oauth) here out of the PR, as it is not directly related. Your suggestion stands, though and needs to be addressed, agreed.